src/Stmt.cc

```diff
@@ -1275,9 +1284,14 @@ ForStmt::ForStmt(IDPList* arg_loop_vars, ExprPtr loop_expr)
 			return;
 			}
 
-		const auto& t = (*loop_vars)[0]->GetType();
+		const auto& lv = (*loop_vars)[0];
+		const auto& t = lv->GetType();
 
-		if ( ! t )
+		if ( lv->IsBlank() )
+			{
+			lv->Error("can not use blank identifier iterating over string");
```

Comment:
Just an observation, the following file somehow leads to wrong diagnostic locations:

```console
$ cat foo.zeek
event zeek_init()
	{
	local _ = "";
	for ( _ in "foo")
		;
	}
$ zeek --parse-only foo.zeek
error in ./foo.zeek, line 3: blank identifier used as local (_)
error in ./foo.zeek, line 3: can not use blank identifier iterating over string (_)
```

I would have expected diagnostics from 2 different lines.